tests/org.eclipse.ui.workbench.texteditor.tests/src/org/eclipse/ui/workbench/texteditor/tests/FindReplaceDialogTest.java

```diff
@@ -1,319 +0,0 @@
-
-		Shell shell= ((Shell)fFindReplaceDialog.get("fActiveShell"));
-		if (shell == null && Util.isGtk()) {
-			if (ScreenshotTest.isRunByGerritHudsonJob()) {
-				takeScreenshot();
-				return;
-			} else
-				fail("this test does not work on GTK unless the runtime workbench has focus. Screenshot: " + takeScreenshot());
-		}
-
-		if (Util.isMac())
-			/* On the Mac, checkboxes only take focus if "Full Keyboard Access" is enabled in the System Preferences.
-			 * Let's not assume that someone pressed Ctrl+F7 on every test machine... */
-			return;
-
-		assertTrue(findField.isFocusControl());
-
-		Button wrapSearchBox= (Button)fFindReplaceDialog.get("fWrapCheckBox");
-		wrapSearchBox.setFocus();
-		event.doit= true;
-		findField.traverse(SWT.TRAVERSE_RETURN, event);
-		runEventQueue();
-		assertTrue(wrapSearchBox.isFocusControl());
-
-		Button allScopeBox= (Button)fFindReplaceDialog.get("fGlobalRadioButton");
-		allScopeBox.setFocus();
-		event.doit= true;
-		findField.traverse(SWT.TRAVERSE_RETURN, event);
-		runEventQueue();
-		assertTrue(allScopeBox.isFocusControl());
-	}
-
-	private String takeScreenshot() {
-		return ScreenshotTest.takeScreenshot(FindReplaceDialogTest.class, testName.getMethodName(), System.out);
-	}
-
-	@Test
-	public void testFocusNotChangedWhenButtonMnemonicPressed() {
-		if (Util.isMac())
-			return; // Mac doesn't support mnemonics.
-
-		openTextViewerAndFindReplaceDialog();
-
-		Combo findField= (Combo)fFindReplaceDialog.get("fFindField");
-		findField.setText("line");
-		final Event event= new Event();
-
-		runEventQueue();
-		Shell shell= ((Shell)fFindReplaceDialog.get("fActiveShell"));
-		if (shell == null && Util.isGtk())
-			if (ScreenshotTest.isRunByGerritHudsonJob()) {
-				takeScreenshot();
-				return;
-			} else
-				fail("this test does not work on GTK unless the runtime workbench has focus. Screenshot: " + takeScreenshot());
-
-		Button wrapSearchBox= (Button)fFindReplaceDialog.get("fWrapCheckBox");
-		wrapSearchBox.setFocus();
-		event.detail= SWT.TRAVERSE_MNEMONIC;
-		event.character= 'n';
-		event.doit= false;
-		wrapSearchBox.traverse(SWT.TRAVERSE_MNEMONIC, event);
-		runEventQueue();
-		assertTrue(wrapSearchBox.isFocusControl());
-
-		Button allScopeBox= (Button)fFindReplaceDialog.get("fGlobalRadioButton");
-		allScopeBox.setFocus();
-		event.detail= SWT.TRAVERSE_MNEMONIC;
-		event.doit= false;
-		allScopeBox.traverse(SWT.TRAVERSE_MNEMONIC, event);
-		runEventQueue();
-		assertTrue(allScopeBox.isFocusControl());
-
-		event.detail= SWT.TRAVERSE_MNEMONIC;
-		event.character= 'r';
-		event.doit= false;
-		allScopeBox.traverse(SWT.TRAVERSE_MNEMONIC, event);
-		runEventQueue();
-		assertTrue(allScopeBox.isFocusControl());
-	}
-
-	@Test
-	public void testShiftEnterReversesSearchDirection() {
-		openTextViewerAndFindReplaceDialog();
-
-		Combo findField= (Combo)fFindReplaceDialog.get("fFindField");
-		findField.setText("line");
-		IFindReplaceTarget target= (IFindReplaceTarget)fFindReplaceDialog.get("fTarget");
```

Comment:
Although the design of these tests is questionable, they still test some of the UI functionality and can still be run against the refactored implementation. Correting one bug is in the `HistoryStore` will make two of the tests work again. This reflective access to get the target can easily be rewritten to extract the `findReplaceLogic` field from the dialog, which then provides access to the target. The test will then fail, because the "Shift+Enter" handling in the dialog seems to be broken due to the refactoring. This should be fixed as well.

Reply:
I have restored the tests in the correct Folder, I have fixed the reflection-parts.

https://github.com/eclipse-platform/eclipse.platform.ui/pull/1132/files#diff-fba7c43fa2af006dd7d8d981b454e5bfef3a7f622b9cc19c09f7432684e01d81R78-R83